### PR TITLE
Fix archive file handling, add more messaging

### PR DESCRIPTION
**Pull request by fordantitrust**
Dear Mike McQuaid,

I've used your original source, it's great. But there are found some small error. I fixed it and send it back for you.

Change logs
- Add more progress status
- Add new option "part" to support multiple tweet.js part
- Fix parsing tweet.js matched to newest version

### CODE REVIEW

File: twitter_delete.rb [hunk 1]
```diff
@@ -20,6 +20,7 @@
   opt :olds, "Keep tweets/likes more than this many days old", default: 9999
   opt :rts, "Keep tweet with this many retweets", default: 5
   opt :favs, "Keep tweets with this many likes", default: 5
+  opt :part, "Keep tweets with this many likes", default: 0
```
Comment (MikeMcQuaid) [inline twitter_delete.rb:23]: This text is the same. What is this option meant to do?

File: twitter_delete.rb [hunk 2]
```diff
@@ -82,10 +83,11 @@ def api_call(method, *args)
 total_likes = [user.favorites_count, MAX_API_TWEETS].min
 oldest_likes_page = (total_likes / MAX_LIKES_PER_PAGE).ceil
 
-oldest_likes_page.downto(1) do |page|
+oldest_likes_page.downto(1) do |page|
```
Comment (MikeMcQuaid) [inline twitter_delete.rb:86]: Remove this trailing space

File: twitter_delete.rb [hunk 3]
```diff
@@ -96,22 +98,33 @@ def api_call(method, *args)
   tweets_to_delete += tweets.reject(&method(:too_new_or_popular?))
 end
 
+puts "==> Checking timeline to API Completed"
```
Comment (MikeMcQuaid) [inline twitter_delete.rb:101]: This message seems wrong

File: twitter_delete.rb [hunk 4]
```diff
@@ -96,22 +98,33 @@ def api_call(method, *args)
...
-  puts "==> Checking archive JS..."
+  puts "==> Checking archive JS part = ".concat(@options[:part].to_s)
+  
   archive_tweet_ids = []
 
   # tweet.js is not valid JSON...
+  puts "==> Fixing tweet.js is not valid JSON"
   file_contents = File.read(@options[:archive])
-  file_contents.sub! "window.YTD.tweet.part0 = ", ""
+  file_contents.sub! "window.YTD.tweet.part".concat(@options[:part].to_s).concat(" = "), ""
```
Comment (MikeMcQuaid) [inline twitter_delete.rb:111]: Can you explain this a bit more?

File: twitter_delete.rb [hunk 5]
```diff
@@ -96,22 +98,33 @@ def api_call(method, *args)
...
+  tweets = JSON.parse(file_contents)
 
-  archive_tweet_ids.each_slice(MAX_TWEETS_PER_REQUEST) do |tweet_ids|
+  puts "==> Found #{tweets.size} tweets in archive data"
+  
+  tweets.each do |tweet|    
+    archive_tweet_ids << tweet["tweet"]["id_str"]
+  end  
+
+  puts "==> Checking archive to API..."
```
Comment (MikeMcQuaid) [inline twitter_delete.rb:122]: Can you explain this? It doesn't make sense as-is.

**Review by MikeMcQuaid — COMMENTED** [review]
Thanks for the PR! Some comments.

File: twitter_delete.rb (hunk 4 above)
Comment (fordantitrust) [reply twitter_delete.rb:111]: Refer from my tweets archive files, it's large lists of tweets get split up into multiple files, which are all found in the same place.  

So as well as "tweet.js" (window.YTD.tweet.part0) I might have "tweet-part1.js" (window.YTD.tweet.part1) , "tweet-part2.js" (window.YTD.tweet.part2) etc.

File: twitter_delete.rb (hunk 3 above)
Comment (fordantitrust) [reply twitter_delete.rb:101]: I removed

File: twitter_delete.rb (hunk 2 above)
Comment (fordantitrust) [reply twitter_delete.rb:86]: I removed

File: twitter_delete.rb (hunk 1 above)
Comment (fordantitrust) [reply twitter_delete.rb:23]: I mean to "Refer to window.YTD.tweet.part[x]" in tweet-part[x].js files. 

Refer from my tweets archive files, it's large lists of tweets get split up into multiple files, which are all found in the same place. 

So as well as "tweet.js" you might have "tweet-part1.js" (window.YTD.tweet.part1) , "tweet-part2.js" (window.YTD.tweet.part2) etc. 

Sorry I'm not edit this label before pull request to you.

Example command
./twitter_delete.rb --user Username --olds 9999 --days 28 --rts 99 --favs 99 --archive /mnt/d/tweet-part1.js --part 1

File: twitter_delete.rb (hunk 5 above)
Comment (fordantitrust) [reply twitter_delete.rb:120]: After fix not valid JSON and parsed json. I'm giving it for report status checking archive from twitter API.

File: twitter_delete.rb (hunk 4 above)
Comment (MikeMcQuaid) [reply twitter_delete.rb:109]: Ok, makes sense. I think ideally this wouldn't require input from the user but instead would read and iterate through all these values. How about:

```suggestion
  file_contents.gsub! /window\.YTD\.tweet\.part\d+ = /, ""
```

File: twitter_delete.rb [hunk 6]
```diff
@@ -20,6 +20,7 @@
   opt :olds, "Keep tweets/likes more than this many days old", default: 9999
   opt :rts, "Keep tweet with this many retweets", default: 5
   opt :favs, "Keep tweets with this many likes", default: 5
+  opt :part, "Refer to window.YTD.tweet.part[x] in tweet.js file", default: 0
```
Comment (MikeMcQuaid) [inline twitter_delete.rb:23]:
```suggestion
```

File: twitter_delete.rb [hunk 7]
```diff
@@ -97,21 +99,30 @@ def api_call(method, *args)
 end
 
 if @options[:archive_given]
-  puts "==> Checking archive JS..."
+  puts "==> Checking archive JS part = ".concat(@options[:part].to_s)
+
```
Comment (MikeMcQuaid) [inline twitter_delete.rb:103]:
```suggestion
  puts "==> Checking archive JS..."
```

File: twitter_delete.rb [hunk 8]
```diff
@@ -86,6 +87,7 @@ def api_call(method, *args)
   tweets = api_call :favorites, count: MAX_LIKES_PER_PAGE, page: page
   tweets_to_unlike += tweets.reject(&method(:too_new?))
 end
+puts "==> Checking likes to API Completed"
```
Comment (MikeMcQuaid) [inline twitter_delete.rb:90]: This is printed after checking but, if you want to print this, should be before.

File: twitter_delete.rb (hunk 5 above)
Comment (MikeMcQuaid) [reply twitter_delete.rb:120]: How about:

```suggestion
  puts "==> Querying archived tweets from API..."
```

File: twitter_delete.rb [hunk 9]
```diff
@@ -97,21 +99,30 @@ def api_call(method, *args)
...
+    archive_tweet_ids << tweet["tweet"]["id_str"]
+  end  
 
-  archive_tweet_ids.each_slice(MAX_TWEETS_PER_REQUEST) do |tweet_ids|
+  puts "==> Checking archive to API..."
+  archive_tweet_ids.each_slice(MAX_TWEETS_PER_REQUEST) do |tweet_ids|    
     tweets = api_call :statuses, tweet_ids
     tweets_to_delete += tweets.reject(&method(:too_new_or_popular?))
   end
+
```
Comment (MikeMcQuaid) [inline twitter_delete.rb:125]:
```suggestion
```

File: twitter_delete.rb (hunk 1 above)
Comment (MikeMcQuaid) [reply twitter_delete.rb]: @fordantitrust Gotcha. I think ideally this would be able to take all archive files, perhaps if passed a folder? Thoughts?

**Comment by MikeMcQuaid** [comment]
@fordantitrust Any thoughts on my comments here? Thanks!

**Pull request closed without merging**